### PR TITLE
5566 doc edit usability

### PR DESCRIPTION
**Pull request by smithellis**
**Requires 1234 Be Pushed First - based on that**
* Edit Article Description, on save, now takes you to article vs. edit page
* "Comment:" label and textarea now hidden on load, shown when checkbox clicked
* textarea reasonably sized

### CODE REVIEW

File: kitsune/sumo/static/sumo/js/wiki.js
Comment (escattone) [inline kitsune/sumo/static/sumo/js/wiki.js:424]: I like `show()` better than `slideDown()`, but that's just my opinion.
```suggestion
        $commentlabel.show();
        $comment.show();
```

**Review by escattone — COMMENTED** [review]
I'll wait until https://github.com/mozilla/kitsune/pull/5447 is merged and this PR is rebased for a final review, but it looks good.

File: kitsune/wiki/views.py
Comment (akatsoulas) [inline kitsune/wiki/views.py:487]: Can't we replace this block with something like:

```python
    rev_form = RevisionForm(
        request.POST or None, instance=rev, initial={"based_on": rev.id, "comment": ""}
    )
    if rev_form.is_valid():
        rev_form.instance.doc = doc
        _document_lock_clear(doc.id, user.username)
        _save_rev_and_notify(rev_form, user, doc, base_rev=rev)
        if "notify-future-changes" in request.POST:
            EditDocumentEvent.notify(request.user, doc)
        return HttpResponseRedirect(reverse("wiki.document_revisions", args=[document_slug]))

    show_revision_warning = _show_revision_warning(doc, rev)
    locked, locked_by = _document_lock(doc.id, user.username)

    return render(
        request,
        "wiki/edit.html",
        {
            "revision_form": rev_form,
            "document": doc,
            "show_revision_warning": show_revision_warning,
            "locked": locked,
            "locked_by": locked_by,
        },
    )
```

**Review by akatsoulas — CHANGES_REQUESTED** [review]

**Comment by akatsoulas** [comment]
I would propose to merge the two PR's into a single one or at least close this one, wait for the other to be merged and change anything that needs changing after the merge. I was working in parallel with the browser and my editor and it took me a while to realize that I am looking two different pull requests. Some if not all of the comments that I made here apply to the other PR. Trying to keep the two PRs in sync is prone to errors.

**Comment by smithellis** [comment]
Closing in favor of https://github.com/mozilla/kitsune/pull/5447

File: kitsune/wiki/views.py
Comment (smithellis) [reply kitsune/wiki/views.py:487]: Your version breaks tests - because it is behaving differently than the original code. In your code, on a POST request, `instance` and `initial` aren't discarded when the form is re-instantiated with POST data. So when `is_valid()` is called, the POST data is validated against the model `instance` constraints and the `initial` data.

File: kitsune/sumo/static/sumo/js/wiki.js
Comment (Tone100) [reply kitsune/sumo/static/sumo/js/wiki.js:424]: Why is this open? Who broke into this, hey man, dont listen to questions and replies from here. This isnt there isp or device.

**Pull request closed without merging**
